project/MimaFilters.scala

```diff
@@ -41,6 +41,21 @@ object MimaFilters extends AutoPlugin {
 
     // PR 10406
     ProblemFilters.exclude[DirectMissingMethodProblem]("scala.reflect.runtime.JavaUniverse#PerRunReporting.deprecationWarning"),
+
+    // extend AbstractIterator
+    ProblemFilters.exclude[MissingTypesProblem]("scala.collection.immutable.MapKeyIterator"),
+    ProblemFilters.exclude[MissingTypesProblem]("scala.collection.immutable.MapKeyValueTupleHashIterator"),
+    ProblemFilters.exclude[MissingTypesProblem]("scala.collection.immutable.MapKeyValueTupleIterator"),
+    ProblemFilters.exclude[MissingTypesProblem]("scala.collection.immutable.MapKeyValueTupleReverseIterator"),
+    ProblemFilters.exclude[MissingTypesProblem]("scala.collection.immutable.MapNodeRemoveAllSetNodeIterator"),
+    ProblemFilters.exclude[DirectMissingMethodProblem]("scala.collection.immutable.MapNodeRemoveAllSetNodeIterator.next"),
+    ProblemFilters.exclude[MissingTypesProblem]("scala.collection.immutable.MapValueIterator"),
+    ProblemFilters.exclude[MissingTypesProblem]("scala.collection.immutable.NewVectorIterator"),
+    ProblemFilters.exclude[MissingTypesProblem]("scala.collection.immutable.SetHashIterator"),
+    ProblemFilters.exclude[MissingTypesProblem]("scala.collection.immutable.SetIterator"),
+    ProblemFilters.exclude[MissingTypesProblem]("scala.collection.immutable.SetReverseIterator"),
+    ProblemFilters.exclude[MissingTypesProblem]("scala.io.BufferedSource"),
+    ProblemFilters.exclude[MissingTypesProblem]("scala.io.Source"),
```

Comment:
The last two are the only non-private, the mima message is `Missing types {scala.collection.AbstractIterator}` (forwards bin compat).

I'd say it's fine, but if we want to be cautious we can skip the change on `abstract class Source`.

Reply:
Out of extra abundance of caution, I reverted both Source and BufferedSource. (Arguably `scala.io` doesn't deserve attention.)